modules/tracking/include/opencv2/tracking/kalman_filters.hpp

```diff
@@ -0,0 +1,215 @@
+#pragma once
+#include "opencv2/core.hpp"
+#include <limits>
+
+namespace cv
+{
+
+/** @brief Cholesky decomposition
+
+ The function performs Cholesky decomposition <https://en.wikipedia.org/wiki/Cholesky_decomposition>.
+ @param A - the Hermitian, positive-definite matrix,
+ @param astep - size of row in A,
+ @param asize - number of cols and rows in A,
+ @param L - the lower triangular matrix, A = L*Lt.
+*/
+template<typename _Tp> bool
+choleskyDecomposition( const _Tp* A, size_t astep, const int asize, _Tp* L )
+{
+    int i, j, k;
+    double s;
+    astep /= sizeof(A[0]);
+    for( i = 0; i < asize; i++ )
+
+    {
+        for( j = 0; j < i; j++ )
+        {
+            s = A[i*astep + j];
+            for( k = 0; k < j; k++ )
+                s -= L[i*astep + k]*L[j*astep + k];
+            L[i*astep + j] = (_Tp)(s/L[j*astep + j]);
+        }
+        s = A[i*astep + i];
+        for( k = 0; k < i; k++ )
+        {
+            double t = L[i*astep + k];
+            s -= t*t;
+        }
+        if( s < std::numeric_limits<_Tp>::epsilon() )
+            return false;
+        L[i*astep + i] = (_Tp)(std::sqrt(s));
+    }
+
+   for( i = 0; i < asize; i++ )
+       for( j = i+1; j < asize; j++ )
+       {
+           L[i*astep + j] = 0.0;
+       }
+
+    return true;
+}
+
+/** @brief The interface for Unscented Kalman filter and Augmented Unscented Kalman filter.
+*/
+class CV_EXPORTS_W UnscentedKalmanFilterInterface
```

Comment:
do not use suffix `Interface`. Instead, use suffix `Impl` in the implementation class